Tests/UnitTests/Caching/PostedTransactionCacheTests.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PostedTransactionCacheTests.swift
+//
+//  Created by Nacho Soto on 7/28/23.
+
+import Foundation
+import Nimble
+import XCTest
+
+@testable import RevenueCat
+
+class PostedTransactionCacheTests: TestCase {
+
+    private var userDefaults: UserDefaults!
+    private var deviceCache: DeviceCache!
+    private var cache: PostedTransactionCache!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        self.userDefaults = try XCTUnwrap(.init(suiteName: UUID().uuidString))
+        self.deviceCache = .init(sandboxEnvironmentDetector: MockSandboxEnvironmentDetector(isSandbox: true),
+                                 userDefaults: self.userDefaults)
+        self.cache = .init(deviceCache: self.deviceCache)
```

Comment:
I decided to test with with real `DeviceCache` and `UserDefaults` which is a lot more accurate and less painful.